proposals/NNNN-add-generic-hmi-plugin-support.md

```diff
@@ -0,0 +1,362 @@
+- Parameters: `appID`
+
+**BUTTON_PRESS**
+
+- Sender: Buttons
+- Receiver: RC
+- Description: Sent when a button related to RC is pressed, informing the RC component to update its data where appropriate.
+- Parameters: `buttonName`
+
+**DEACTIVATE_APP**
+
+- Sender: Various
+- Receiver: Main HMI
+- Description: Sent when an HMI event (such as EMERGENCY_EVENT) which should disable the active app is started. Closes the active app and returns to the app screen in the main HMI.
+- Parameters: `appID`
+
+**SEND_NAV_CAPABILITIES**
+
+- Sender: Navigation
+- Receiver: Main HMI
+- Description: Sent when the navigation plugin is registered, communicating its capabilities to be sent in `UI.GetCapabilities`.
+- Parameters: `navigationCapabilities`
+
+**SPEAK**
+
+- Sender: Various
+- Receiver: TTS
+- Description: Sent when any TTS prompt is triggered outside of the TTS interface (ex. `helpPrompt`/`timeoutPrompt` in VR.PerformInteraction). The TTS plugin will speak whatever data is provided.
+- Parameters: `ttsChunks`
+
+**ALERT_MANEUVER**
+
+- Sender: Navigation
+- Receiver: UI
+- Description: Sent whenever a Navigation.AlertManeuver request is received by the Navigation plugin. The UI should display an alert maneuver popup with the provided softbuttons.
+- Parameters: `softButtons`, `appID`
+
+**SETTINGS_UPDATE**
+
+- Sender: Settings
+- Receiver: Main HMI
+- Description: Sent whenever any HMI settings are changed in the Settings plugin.
+- Parameters: `hmiTheme`, etc.
+
+**POPUP**
+
+- Sender: Various
+- Receiver: Main HMI
+- Description: Sent whenever a plugin needs to display a prompt to the user (ex. `RC.GetInteriorVehicleDataConsent` or `AppService.GetActiveServiceConsent`). Response is sent when an option is selected.
+- Parameters: `text`, `softButtons`
+- Response Parameters: `buttonID`
+
+**Note:** This is not a comprehensive list, other interactions between components may need to be added as necessary during implementation. In addition, customized messages will likely need to be created for production systems to accomodate differences in plugin design.
+
+### Integrated Plugin Controls
+
+While a majority of the functionality for these plugins can be handled in the background, production implementations will need the ability to integrate some portion of these plugins directly into the Generic HMI (such as a set of controls/settings for the plugin). To accomodate this, stubs for these integrated components can be included in the project and these can be expanded to fit a specific system.  
+
+![Plugin UI Stub](../assets/proposals/NNNN-add-generic-hmi-plugin-support/vr-plugin-stub-example.png)
+
+To accomodate this, we will need to include a way to navigate between various screens in the Generic HMI. This could potentially be done via a persistent menu created using existing node packages such as [react-burger-menu](https://github.com/negomi/react-burger-menu).
```

Comment:
```suggestion
To accommodate this, we will need to include a way to navigate between various screens in the Generic HMI. This could potentially be done via a persistent menu created using existing node packages such as [react-burger-menu](https://github.com/negomi/react-burger-menu).
```